### ISSUE TITLE
Improve cli help documents and command line help

### ISSUE THREAD

**Issue by chrislovecnm** [kubernetes/kops#1005]
Appropriate the https://github.com/kubernetes/kubernetes/tree/master/pkg/kubectl/cmd/templates code so that we can do cool stuff like:

```golang
var (
	apply_long = templates.LongDesc(`
		Apply a configuration to a resource by filename or stdin.
		This resource will be created if it doesn't exist yet.
		To use 'apply', always create the resource initially with either 'apply' or 'create --save-config'.

		JSON and YAML formats are accepted.

		Alpha Disclaimer: the --prune functionality is not yet complete. Do not use unless you are aware of what the current state is. See https://issues.k8s.io/34274.`)

	apply_example = templates.Examples(`
		# Apply the configuration in pod.json to a pod.
		kubectl apply -f ./pod.json

		# Apply the JSON passed into stdin to a pod.
		cat pod.json | kubectl apply -f -`)
)

func NewCmdApply(f cmdutil.Factory, out io.Writer) *cobra.Command {
	var options ApplyOptions

	cmd := &cobra.Command{
		Use:     "apply -f FILENAME",
		Short:   "Apply a configuration to a resource by filename or stdin",
		Long:    apply_long,
		Example: apply_example,
		Run: func(cmd *cobra.Command, args []string) {
			cmdutil.CheckErr(validateArgs(cmd, args))
			cmdutil.CheckErr(cmdutil.ValidateOutputArgs(cmd))
			cmdutil.CheckErr(validatePruneAll(options.Prune, cmdutil.GetFlagBool(cmd, "all"), options.Selector))
			cmdutil.CheckErr(RunApply(f, cmd, out, &options))
		},
	}
}
```

**Comment by chrislovecnm** [comment]
@robertojrojas if you could update this with a couple of use cases and a quick design in a PR that would be an awesome place to start.

**Comment by kris-nova** [comment]
Please - can we look at getting  a sweet ascii logo 

https://github.com/kubernetes/kops/issues/795

**Comment by robertojrojas** [comment]
I think we should start by migrating the root command to follow the same approach
used by k8s/kubectl, similar to the example by @chrislovecnm  at the beginning of this issue.
We can leverage the fact that project is already vendoring k8s and use 
the same code in kubectl/cmd/templates to group and template commands. 
This should also provide an easier way to follow changes made by k8s in terms of commands, 
help, etc...
There is one extra  dependency to vendor "github.com/MakeNowJust/heredoc" related to templates.

questions:

- Should we classify commands in a similar manner as k8s/kubectl in terms of level? 
       Basic Commands (Beginner), Basic Commands (Intermediate), etc....

- Should we split this issue into one issue per command to migrate? The idea behind this
      is to make it easier for PR reviews and merge. We can alway start with 
      this issue covering the root command and add other issues for rest of the commands.

- @kris-nova I like the logo idea. Always liked that when starting the Redis Server.  Will that be part of #795 or it's own issue?

**Comment by chrislovecnm** [comment]
questions:

> Should we classify commands in a similar manner as k8s/kubectl in terms of level?
Basic Commands (Beginner), Basic Commands (Intermediate), etc....

We don't really have those yet, in my mind.  Maybe @kris-nova has other ideas.

> Should we split this issue into one issue per command to migrate? The idea behind this
is to make it easier for PR reviews and merge. We can alway start with
this issue covering the root command and add other issues for rest of the commands.

root and start with the more popular items, create, delete, edit, update.

**Comment by kris-nova** [comment]
> We don't really have those yet, in my mind. Maybe @kris-nova has other ideas.

Lets keep it simple for now. We can always add more features in later.

---

Another point I wanted to mention was the shorthand description for each command desperately needs to be updated in layman. Right now the description for commands like `kops delete cluster` just says `delete cluster` and that feels a bit wrong, and un informative.

**Comment by chrislovecnm** [comment]
In master this is now updated with the internationalization stuff as well. Would be awesome to have kops help in Hebrew ;)
